flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -102,4 +116,18 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 	public void cancel() {
 		running = false;
 	}
+
+	@Override
+	protected void acknowledgeIDs(List<Long> ids) {
+		try {
+			channel.basicAck(ids.get(ids.size() - 1), true);

```

Comment:
Since we acknowledge all ids <= the last stored id (the true flag), it is sufficient to store only the last id.